src/formElementPlugins/search/src/useDebounce.ts

```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from 'react'
+
+// Adapted from:
+// https://typeofnan.dev/writing-a-custom-react-usedebounce-hook-with-typescript/
+
+export default function useDebounce<T>(initialValue: T, delay: number): [T, React.Dispatch<T>] {
+  const [debounceInput, setDebounceInput] = useState<T>(initialValue)
+  const [debounceOutput, setDebounceOutput] = useState<T>(initialValue)
+
+  useEffect(() => {
+    const debounce = setTimeout(() => {
+      setDebounceOutput(debounceInput)
+    }, delay)
+    return () => {
+      clearTimeout(debounce)
+    }
+  }, [debounceInput])
+
+  return [debounceOutput, setDebounceInput]
```

Comment:
There seems to be a problem with that - when using UC API and searching for the 2nd drug, it doesn't stop trying to fetch data and show is "loading" for ever: as pointed out [here](https://github.com/openmsupply/application-manager-server/pull/380#discussion_r642689412)

Reply:
Great catch, this one -- it only happens if you enter the same search term immediately after selecting a result.

Turns out if the input is the same, the useEffect for `debounceOutput` never runs as the value of `debounceOutput` doesn't change.

Just needed to reset `debounceInput` after the search finishes.   43d00cfd

Reply:
Nice : ) 👍